archfx_cloud/api/connection.py

```diff
@@ -366,25 +289,24 @@ def refresh_token(self):
         :return: True if token was refreshed. False otherwise
         """
         assert self.token_type == DEFAULT_TOKEN_TYPE
-        url = '{0}/{1}'.format(self.base_url, 'auth/api-jwt-refresh/')
-
-        payload = json.dumps({'token': self.token})
+        if self.refresh_token_data:
+            data = {"refresh": self.refresh_token_data}
```

Comment:
This is technically not needed as `auth/api-jwt-refresh` is a custom method, so we can take whatever we want and convert to what the library requires on the cloud/broker side.

But the real issue here is that during migration, we may need to deploy two APIs side by side, so we may need to send to a different end point depending on the package. But that will only be for a few days so maybe we just ignore that.

Reply:
The main idea is this client implementation supports both old and new REST JWT styles, transparently to the user.

The central cloud instance can remain on `djangorestframework-jwt==1.11.0` until you want to upgrade to Django 3.2 (`djangorestframework-jwt` pins `djangorestframework` to version 3.11, (which already doesn't support Django 3.1, but still works, warning about that it will break on Django 3.2) - giving us an almost infinite migration window.

`refresh_token_data` being there means that `SimpleJWT` is used (because `SimpleJWT` returns `access` and `refresh` fields in payload instead of `token` and that's how module learns what kind of server it's talking to).
Since "refresh token" views are part of JWT plugin and expect different fields to be in JSON - client asks for a refresh in a format the server will understand.

Re: customers using the API - are we talking about on-premise servers or the cloud server?
Because I just found 4 different implementations of ArchFX cloud client in `archfx_broker` repo and going to unify those.

Re:  https://github.com/dkarchmer/django-rest-framework-client - is it still used anywhere?

Reply:
Yes, as I said, I am ok with this changes just to unblock the broker, but I also don't see the value of cleaning up the broker and not the cloud. All I am saying is that we are not done until everything is upgrading, so upgrading the broker is actually fairly low priority. But yes, if for whatever reason, you cannot wait, I am ok with these changes as temporary changes.

But please note that the Broker and Cloud must have "identical" implementations when we are all set and done, so if the Cloud forces us into any type of spec, we will keep that same spec on the Broker. It is critical we use the identical models and API between Broker and Cloud

Reply:
It's not fully possible due to `MachineAPI` part of the Broker needing "admin" tokens (and not being in the cloud at all), but it can be made close enough. The change is about 20 lines of code. I'll put up a PR for that tomorrow, you'll see how easy it is :)